### PR TITLE
note_input_optimizations

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Resolves: https://github.com/musescore/MuseScore/issues/16112
Resolves: https://github.com/musescore/MuseScore/issues/21520
Resolves: https://github.com/musescore/MuseScore/issues/22639

Muse Sounds note: to hear the difference, you need MuseSampler 0.100, which hasn't been publically released yet but will be with MuseScore Studio 4.4

### CODE REVIEW

**Comment by metasekk** [comment]
(I know that it is still a draft, and that I have no legitimacy here, but just to mention in case you would have missed it: after exporting scores as audio with Saturday's build - the latest as for now -, Muse Sound staves are silent in the exported mp3 file. For reference, on my side, everything worked as expected in Tuesday's https://github.com/musescore/MuseScore/pull/22791#commits-pushed-80c4f56 build)

**Comment by RomanPudashkin** [comment]
> (I know that it is still a draft, and that I have no legitimacy here, but just to mention in case you would have missed it: after exporting scores as audio with Saturday's build - the latest as for now -, Muse Sound staves are silent in the exported mp3 file. For reference, on my side, everything worked as expected in Tuesday's [#22791 (commits)](https://github.com/musescore/MuseScore/pull/22791#commits-pushed-80c4f56) build)

Thank you! I'll investigate...

File: src/framework/musesampler/internal/musesamplerwrapper.cpp [hunk 1]
```diff
@@ -45,6 +45,10 @@ MuseSamplerWrapper::MuseSamplerWrapper(MuseSamplerLibHandlerPtr samplerLib,
     m_sequencer.setOnOffStreamFlushed([this]() {
         revokePlayingNotes();
     });
+
+//    config()->samplesToPreallocateChanged().onReceive(this, [this](const samples_t samples) {
```
Comment (Eism) [inline src/framework/musesampler/internal/musesamplerwrapper.cpp:49]: is it still needed?

**Review by Eism — APPROVED** [review]

File: src/framework/musesampler/internal/musesamplerwrapper.cpp (hunk 1 above)
Comment (RomanPudashkin) [reply src/framework/musesampler/internal/musesamplerwrapper.cpp:49]: Yes, there is a bug on the MuseSampler side, we're working on it. This code allows us to test it

**Review by Eism — APPROVED** [review]

**Comment by bkunda** [comment]
I'm happy for this to be merged in its current state as it will enable broader testing and easier build management, and will unblock the creation of the 4.4 branch. 
Thank you @RomanPudashkin!

**Comment by DmitryArefiev** [comment]
Tested on Win10, Mac13.6, LinuxUbuntu22.04 - **PASS*
On Linux,  MuseSampler 0.100 doesn't work with MuseHub2 for now

**Pull request merged**